http_types/utils.py

```diff
@@ -59,10 +67,22 @@ def __init__(self):
     def from_dict(obj: Dict) -> Request:
         obj_copy = dict(**obj)
 
+        if not "path" in obj_copy:
+            if not "pathname" in obj_copy:
+                raise Exception("One of 'path' or 'pathname' is required")
+            path = obj_copy["pathname"]
+            if "query" in obj_copy:
+                path += "?" + urlencode(obj_copy["query"])
+            obj_copy['path'] = path
+
         if not "pathname" in obj_copy:
             path = obj_copy['path']
             obj_copy['pathname'] = parse_pathname(path)
 
+        if not "query" in obj_copy and "path" in obj_copy:
```

Comment:
I assume it should be an error if `path` is not in `obj_copy` at this point?

Reply:
In the case of no query parameters, I think it's ok to have only `pathname` but no `query` property.